src/parser/jobs/brd/PitchPerfectUsage.js

```diff
@@ -0,0 +1,314 @@
+			get stacks() { return this.damageEvent && this.damageEvent.stacks || undefined },
+			get timestamp() { return this.damageEvent && this.damageEvent.timestamp },
+		}
+
+		this._ppEvents.push(ppEvent)
+
+		// Only an issue if there are dot ticks left on the song and sufficient time to use PP (animation lock)
+		if (enemy.lastTick + DOT_TICK_FREQUENCY < wm.timestamp + SONG_DURATION - ANIMATION_LOCK) {
+
+			// We write down the crit on each dot, to provide the information later
+			ppEvent.critOnDot[STATUSES.CAUSTIC_BITE.id] = enemy.tick[STATUSES.CAUSTIC_BITE.id] && enemy.tick[STATUSES.CAUSTIC_BITE.id].expectedCritRate * CONVERSION_FACTOR
+			ppEvent.critOnDot[STATUSES.STORMBITE.id] = enemy.tick[STATUSES.STORMBITE.id] && enemy.tick[STATUSES.STORMBITE.id].expectedCritRate * CONVERSION_FACTOR
+
+			// If crit is above threshold for PP2
+			// Using PP1 when not at the end of the song is not ideal
+			if (
+				enemy.tick[STATUSES.CAUSTIC_BITE.id]
+				&& enemy.tick[STATUSES.STORMBITE.id]
+				&& enemy.tick[STATUSES.CAUSTIC_BITE.id].expectedCritRate * CONVERSION_FACTOR > PP2_THRESHOLD
+				&& enemy.tick[STATUSES.STORMBITE.id].expectedCritRate * CONVERSION_FACTOR > PP2_THRESHOLD
+			) {
+				// Using PP1 when not at the end of the song is not ideal
+				if (event.stacks === PP[1]) {
+					ppEvent.issue = PP1_NOT_AT_END
+				}
+			// Using PP2 when crit is below the threshold is not ideal
+			} else if (event.stacks === PP[2]) {
+
+				ppEvent.issue = PP2_ON_LOW_CRIT
+
+			// Using PP1 when not at the end of the song is not ideal
+			} else if (event.stacks === PP[1]) {
+
+				ppEvent.issue = PP1_NOT_AT_END
+
+			}
+		}
+	}
+
+	_onComplete() {
+		const badPPs = this._ppEvents.filter(pp => pp.issue !== NONE).length
+
+		if (badPPs > 0) {
+			this.suggestions.add(new TieredSuggestion({
+				icon: ACTIONS.PITCH_PERFECT.icon,
+				content: <Fragment>
+					Use {ACTIONS.PITCH_PERFECT.name} at <strong>3 stacks</strong>, unless the critical hit rate on your DoTs is greater than <strong>{PP2_THRESHOLD}%</strong>. Only use it at <strong>1 stack</strong> when there are no more DoT ticks before <ActionLink {...ACTIONS.THE_WANDERERS_MINUET} /> ends. More information in the <a href="javascript:void(0);" onClick={() => this.parser.scrollTo(this.constructor.handle)}>{this.constructor.title}</a> module below.
+				</Fragment>,
+				tiers: {
+					8: SEVERITY.MAJOR,
+					5: SEVERITY.MEDIUM,
+					2: SEVERITY.MINOR,
+				},
+				value: badPPs,
+				why: <Fragment>
+					{badPPs} casts of {ACTIONS.PITCH_PERFECT.name} with the wrong amount of stacks
+				</Fragment>,
+			}))
+		}
+	}
+
+	output() {
+		const badPPevents = this._ppEvents.filter(pp => pp.issue !== NONE)
+
+		if (badPPevents.length === 0) {
+			return
+		}
+
+		// Builds a panel for each pp event
+		const panels = badPPevents.map(pp => {
+
+			const panelProperties = {
+				pp: pp,
+				tuples: [],
+			}
+
+			// For each PP issue
+			if (pp.issue === PP2_ON_LOW_CRIT) {
+				panelProperties.tuples.push({
```

Comment:
Would be good to start moving this stuff into i18n-able content soon. Check the `TransMarkdown` component re: formatting for large blocks of text.